test/functional/functional_query_test.rb

```diff
@@ -81,6 +81,15 @@ def test_select_query_with_multibyte_column_names_results
     assert_equal 'ニ', r.row_description[1].name
   end
 
+  # Query with multiple statements is not officailly supported
+  # Results of only the last statement are obtained
+  def test_query_with_multiple_statements
+    r = @connection.query("SET SEARCH_PATH TO default; SELECT 1 as one")
+    assert_equal 1, r.size
+    assert_equal 1, r.row_description.length
+    assert_equal 'one', r.row_description[0].name
+  end
```

Comment:
I changed the test query to `"SET SEARCH_PATH TO default; SELECT 1 as one; SELECT 2 as two"`.